From b80a1f203688d4ce261ec2e041aae1cc28f26a53 Mon Sep 17 00:00:00 2001
From: Tu Vuong <tu.vuong@arm.com>
Date: Tue, 19 Nov 2013 17:19:18 +0000
Subject: [PATCH 08/20] ARM: drm/exynos: re-enable support for triple buffering

This is putting back HardKernel's previous changes to
have a virtual y resolution of 3 times the physical
display resolution.

Signed-off-by: Guillaume Tucker <guillaume.tucker@arm.com>
---
 drivers/gpu/drm/exynos/exynos_drm_fbdev.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/exynos/exynos_drm_fbdev.c b/drivers/gpu/drm/exynos/exynos_drm_fbdev.c
index 19b58d5..c206481 100644
--- a/drivers/gpu/drm/exynos/exynos_drm_fbdev.c
+++ b/drivers/gpu/drm/exynos/exynos_drm_fbdev.c
@@ -25,6 +25,7 @@
 #include "exynos_drm_iommu.h"
 #define IOCTL_GET_FB_DMA_BUF _IOWR('m',0xF9, __u32 )
 #include <linux/dma-buf.h>
+#define NUM_BUFFERS 3
 
 #define MAX_CONNECTOR		4
 #define PREFERRED_BPP		32
@@ -128,11 +129,11 @@ static int exynos_drm_fbdev_update(struct drm_fb_helper *helper,
 	struct fb_info *fbi = helper->fbdev;
 	struct drm_device *dev = helper->dev;
 	struct exynos_drm_gem_buf *buffer;
-	unsigned int size = fb->width * fb->height * (fb->bits_per_pixel >> 3);
+	unsigned int size = fb->width * fb->height * (fb->bits_per_pixel >> 3) * NUM_BUFFERS;
 	unsigned long offset;
 
 	drm_fb_helper_fill_fix(fbi, fb->pitches[0], fb->depth);
-	drm_fb_helper_fill_var(fbi, helper, fb->width, fb->height); 
+	drm_fb_helper_fill_var(fbi, helper, fb->width, fb->height / NUM_BUFFERS);
 
 	/* RGB formats use only one buffer */
 	buffer = exynos_drm_fb_buffer(fb, 0);
@@ -192,7 +193,7 @@ static int exynos_drm_fbdev_create(struct drm_fb_helper *helper,
 			sizes->surface_bpp);
 
 	mode_cmd.width = sizes->surface_width;
-	mode_cmd.height = sizes->surface_height;
+	mode_cmd.height = sizes->surface_height * NUM_BUFFERS;
 	mode_cmd.pitches[0] = sizes->surface_width * (sizes->surface_bpp >> 3);
 	mode_cmd.pixel_format = drm_mode_legacy_fb_format(sizes->surface_bpp,
 							  sizes->surface_depth);
-- 
1.9.1

